lib/glific/templates.ex

```diff
@@ -262,43 +263,48 @@ defmodule Glific.Templates do
   end
 
   @doc false
-  @spec do_update_hsms(map(), Organization.t()) :: :ok
-  def do_update_hsms(templates, organization) do
+  @spec do_update_hsms(list(), Organization.t(), atom()) :: :ok
+  def do_update_hsms(templates, organization, phase \\ :gupshup) do
```

Comment:
phase or bsp ?

Reply:
I think you can create new function and then match the current provider in that. 

Just like optin contacts function.

Reply:
update with name as bsp